### PR TITLE
Event sessions

### PR DESCRIPTION
**Pull request by ARPspoofing**
Completed:
- As a CS2040 tutorial TA, I want to add tutorial events so that I can keep track of students progress
- As a CS2040 tutorial TA, I want to delete tutorial events so that I can remove cancelled tutorial classes

To be completed: 
- As a CS2040 Lab TA, I want to add lab events so that I can keep track of students progress
- As a CS2040 TA, I want to add consultation events so that I can keep track of queries raised by specific students
- As a CS2040 lab TA, I want to delete lab events so that I can remove cancelled lab classes
- As a CS2040 TA, I want to delete consultation events so that I can remove cancelled consultation sessions
- As a CS2040 TA, I want to add notes so that I do not lose track of what I plan to say
- As a CS2040 TA, I want to delete notes so that I can remove what I have already done
- As a CS2040 TA, I want to add files so that I can refer to specific attachments for classes easily
- As a CS2040 TA, I want to remove files so that I can remove wrongly uploaded attachments

### CODE REVIEW

File: src/main/java/seedu/address/model/AddressBook.java [hunk 1]
```diff
@@ -93,11 +102,50 @@ public void removePerson(Person key) {
...
+     * Returns true if a person with the same identity as {@code person} exists in the address book.
+     */
+    public boolean hasTutorial(Tutorial tutorial) {
+        requireNonNull(tutorial);
+        return tutorials.contains(tutorial);
+    }
+
+    /**
+     * Adds a tutorial to the address book.
+     * The tutorial must not already exist in the address book.
```
Comment (ShiJiaAo) [inline src/main/java/seedu/address/model/AddressBook.java:117]: Not sure if it is mandatory for javadocs to contain @param if there are params specified in the function. In this case it is very clear what Tutorial p is, but under https://se-education.org/guides/conventions/java/intermediate.html#comments it does not seem to say that @param can be excluded.. maybe we can check with our tutor on this. We can still merge the PR and add these things later if we need to.

File: src/main/java/seedu/address/model/event/Consultation.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+     * @param student
+     */
+    public void addStudent(Person student) {
+        super.addStudent(student);
+    }
+
+    /**
+     * Removes a specific student from the list of students
+     * @param student
+     */
```
Comment (ShiJiaAo) [inline src/main/java/seedu/address/model/event/Consultation.java:60]: An empty line needs to be included between the description and parameter section according to our style guide!

**Review by ShiJiaAo — APPROVED** [review]
Thanks for this PR Reyaaz, generally LGTM. Just wondering if we can remove the UserGuide.md changes since it may disturb the changes that Chun Kiat has made. Otherwise just a few style concerns with regards to the javadoc comments, but I think we can merge first and slowly edit the javadoc style later if we need to.

File: src/main/java/seedu/address/model/AddressBook.java (hunk 1 above)
Comment (ARPspoofing) [reply src/main/java/seedu/address/model/AddressBook.java:117]: ok, sounds good to me. Let's check with Bryann in the upcoming tutorial. I shall leave it here and merge it into master for now

File: src/main/java/seedu/address/model/event/Consultation.java (hunk 2 above)
Comment (ARPspoofing) [reply src/main/java/seedu/address/model/event/Consultation.java:60]: Ok, I have resolved it

File: docs/UserGuide.md [hunk 3]
```diff
@@ -4,6 +4,8 @@ title: User Guide
 ---
 
 This project is based on the AddressBook-Level3 project created by the [SE-EDU initiative](https://se-education.org).
+AddressBook Level 3 (AB3) is a desktop app for managing contacts, optimized for use via a Command Line Interface (CLI) while still having the benefits of a Graphical User Interface (GUI). If you can type fast, AB3 can get your contact management tasks done faster than traditional GUI apps.
```
[the remark this answers is not in the document]
Comment (ARPspoofing) [reply docs/UserGuide.md:7]: I have merged the PR Chun Kiat created into the UG. So this will not be an issue I think

**Comment by ARPspoofing** [comment]
closes #54

**Pull request merged**